src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -301,11 +277,11 @@ export class DeployCommand implements Command {
       context.telemetry.properties.parameterFileProvided = "true";
     }
 
-    const accessToken: AccessToken = await subscription.credentials.getToken([]);
+    const accessToken: AccessToken = await createSubscriptionContext(subscription).credentials.getToken();
 
     if (accessToken) {
       const token = accessToken.token;
-      const expiresOnTimestamp = String(accessToken.expiresOnTimestamp);
+      const expiresOnTimestamp = accessToken.expiresOnTimestamp ? String(accessToken.expiresOnTimestamp) : undefined;
```

Comment:
Is it possible that `expiresOnTimestamp` is null? This should always be returned by eSTS, otherwise it's a bug.

Reply:
I'm asking because, from a security perspective, the client should avoid overriding the OAuth access token expireOn property. It should be strictly managed by the authorization server.

Reply:
See comments here:

<img width="1239" alt="image" src="https://github.com/user-attachments/assets/a58591be-f4cd-4e92-ad27-dc219ddface6">

Although it probably makes sense to have that code here instead...

Reply:
I see. It’s odd that VS Code doesn’t provide the expiresOn value, especially since it allows extensions to retrieve the access token. In this case, we might need to set expiresOn to a very short duration, such as 5 minutes. In national clouds, access tokens usually expire in 20 minutes rather than an hour. I think this approach is acceptable, as long as we ensure the value remains within the expiration time specified by the authentication server.

Reply:
Yes, it is, although maybe it makes sense to an extent given that they are handling refresh.  Either way, there is a bug: fd9a9a99069cf804136fa7a66de5edbcb6ee49ba

I moved all the code to the extension and improved the comments.